src/main.rs

```diff
@@ -0,0 +1,192 @@
+use std::collections::HashMap;
+use std::time::Duration;
+
+use clap::Parser;
+
+use consul_external_dns::hetzner_dns;
+use tokio::time::sleep;
+
+use consul_external_dns::config::{Config, DnsProvider};
+use consul_external_dns::consul::{ConsulClient, DnsRecord};
+use consul_external_dns::dns_trait::DnsProviderTrait;
+use tokio_util::sync::CancellationToken;
+
+#[tokio::main]
+async fn main() {
+    let cancel = CancellationToken::new();
+    println!("Starting up Consul External DNS");
```

Comment:
We'll want something like [this](https://github.com/svenstaro/wiresmith/blob/6e40b2b8c88d11a953c2111c5421bec734e8bc33/src/main.rs#L19-L29) to gracefully shut down on SIGINT.

Reply:
fixed